### PR TITLE
[PLUGIN-994] Adding setFetchSize to the database plugins

### PR DESCRIPTION
**Pull request by itsankit-google**
[PLUGIN-994](https://cdap.atlassian.net/browse/PLUGIN-994)

### CODE REVIEW

File: mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSource.java [hunk 1]
```diff
@@ -173,6 +174,15 @@ public void validate(FailureCollector collector) {
     public boolean canConnect() {
       return super.canConnect() && !containsMacro(DATABASE);
     }
+
+    @Override
+    public Map<String, String> getConnectionArguments() {
+      Map<String, String> arguments = new HashMap<>(super.getConnectionArguments());
+      // If connected to MySQL > 5.0.2, and setFetchSize() > 0 on a statement,
```
Comment (tivv) [inline mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSource.java]: I think this should go into `getDBSpecificArguments`. Also Please add a check for fetch size. If for whatever reason it's <=0 we should not add this argument

Comment (itsankit-google) [reply mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSource.java]: done

File: database-commons/src/main/java/io/cdap/plugin/db/batch/source/AbstractDBSource.java [hunk 2]
```diff
@@ -471,6 +475,11 @@ public boolean canConnect() {
         !containsMacro(ConnectionConfig.USER) && !containsMacro(ConnectionConfig.PASSWORD) &&
         !containsMacro(DBSourceConfig.DATABASE) && !containsMacro(DBSourceConfig.IMPORT_QUERY);
     }
+
+    @Override
+    public Integer getFetchSize() {
+      return 0;
```
Comment (tivv) [inline database-commons/src/main/java/io/cdap/plugin/db/batch/source/AbstractDBSource.java]: Why don't we add proper fetchSize here?

**Review by tivv — COMMENTED** [review]
Does this cover CloudSQL plugins?

**Comment by tivv** [comment]
I see that CloudSQL plugins descent from io.cdap.plugin.db.batch.source.AbstractDBSource.DBSourceConfig and that one has hard coded "0" as fetch size

File: database-commons/src/main/java/io/cdap/plugin/db/batch/source/AbstractDBSource.java (hunk 2 above)
Comment (itsankit-google) [reply database-commons/src/main/java/io/cdap/plugin/db/batch/source/AbstractDBSource.java]: done

**Comment by itsankit-google** [comment]
Yes, it does cover CloudSQL plugins as discussed offline.

**Review by tivv — APPROVED** [review]

**Pull request merged**